<div align="center"><span style="color: #006600;"><h1> NYC DEATH RATES </h1></span></div>
<br>
<div align="center"><span style="color:green"><h4> Rickardo Henry -- IS362 </h4></span></div>

We will begin by importing pandas and reading the .csv data. The data is captured is from [Github](https://github.com/) from Djamshed Djuraev. We will begin by importing pandas and then tidying up the data.

In [2]:
import pandas as pd # importing pandas module for usage
import numpy as np

deaths = pd.read_csv('C:/Users/rickardo.henry/Desktop/IS362_Project2/NYC-Death-Rates.csv')
# Reading Data from CSV file.
deaths

Year                                      Leading Cause Sex  \
0    2014                        Diabetes Mellitus (E10-E14)   F   
1    2011          Cerebrovascular Disease (Stroke: I60-I69)   M   
2    2008              Malignant Neoplasms (Cancer: C00-C97)   M   
3    2010              Malignant Neoplasms (Cancer: C00-C97)   F   
4    2012          Cerebrovascular Disease (Stroke: I60-I69)   M   
5    2007  Mental and Behavioral Disorders due to Use of ...   M   
6    2011                                   All Other Causes   F   
7    2007       Chronic Lower Respiratory Diseases (J40-J47)   F   
8    2012  Essential Hypertension and Renal Diseases (I10...   F   
9    2009     Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   
10   2011       Chronic Lower Respiratory Diseases (J40-J47)   M   
11   2008          Cerebrovascular Disease (Stroke: I60-I69)   F   
12   2013     Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
13   2010          Cerebrovascular Disease (Stroke: I60-I69)   F   
14   2012     Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
15   2012              Malignant Neoplasms (Cancer: C00-C97)   M   
16   2011          Cerebrovascular Disease (Stroke: I60-I69)   F   
17   2014  Congenital Malformations, Deformations, and Ch...   F   
18   2009                                   All Other Causes   F   
19   2014                        Diabetes Mellitus (E10-E14)   M   
20   2014  Accidents Except Drug Posioning (V01-X39, X43,...   M   
21   2012          Cerebrovascular Disease (Stroke: I60-I69)   F   
22   2009     Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
23   2014  Essential Hypertension and Renal Diseases (I10...   F   
24   2008  Certain Conditions originating in the Perinata...   M   
25   2012  Mental and Behavioral Disorders due to Acciden...   M   
26   2014              Malignant Neoplasms (Cancer: C00-C97)   F   
27   2011  Mental and Behavioral Disorders due to Acciden...   F   
28   2014  Accidents Except Drug Posioning (V01-X39, X43,...   F   
29   2008          Cerebrovascular Disease (Stroke: I60-I69)   M   
..    ...                                                ...  ..   
263  2012                               Septicemia (A40-A41)   F   
264  2014                                   All Other Causes   M   
265  2008                        Diabetes Mellitus (E10-E14)   M   
266  2008  Essential Hypertension and Renal Diseases (I10...   F   
267  2011  Human Immunodeficiency Virus Disease (HIV: B20...   F   
268  2012                        Diabetes Mellitus (E10-E14)   F   
269  2007    Intentional Self-Harm (Suicide: X60-X84, Y87.0)   M   
270  2009     Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
271  2014              Malignant Neoplasms (Cancer: C00-C97)   M   
272  2007                                   All Other Causes   M   
273  2014  Essential Hypertension and Renal Diseases (I10...   M   
274  2010     Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   
275  2008  Human Immunodeficiency Virus Disease (HIV: B20...   M   
276  2010            Influenza (Flu) and Pneumonia (J09-J18)   F   
277  2014     Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   
278  2014              Malignant Neoplasms (Cancer: C00-C97)   M   
279  2009                                   All Other Causes   M   
280  2012            Influenza (Flu) and Pneumonia (J09-J18)   F   
281  2008     Chronic Liver Disease and Cirrhosis (K70, K73)   F   
282  2013  Accidents Except Drug Posioning (V01-X39, X43,...   F   
283  2011       Chronic Lower Respiratory Diseases (J40-J47)   F   
284  2007                        Diabetes Mellitus (E10-E14)   F   
285  2012  Accidents Except Drug Posioning (V01-X39, X43,...   F   
286  2009                                   All Other Causes   M   
287  2013              Malignant Neoplasms (Cancer: C00-C97)   F   
288  2007                        Diabetes Mellitus (E10-E14)   M   
289  2010  Essential Hypertension and Renal Diseases (I10... 

<span style='color:orange'><i> We will now tidy up the data above by replacing all the <b>'.'</b> with 'NaN' values. </i></span>

In [3]:
tidy = deaths.replace(to_replace = '.', value = np.nan)
tidy

Year                                      Leading Cause Sex  \
0    2014                        Diabetes Mellitus (E10-E14)   F   
1    2011          Cerebrovascular Disease (Stroke: I60-I69)   M   
2    2008              Malignant Neoplasms (Cancer: C00-C97)   M   
3    2010              Malignant Neoplasms (Cancer: C00-C97)   F   
4    2012          Cerebrovascular Disease (Stroke: I60-I69)   M   
5    2007  Mental and Behavioral Disorders due to Use of ...   M   
6    2011                                   All Other Causes   F   
7    2007       Chronic Lower Respiratory Diseases (J40-J47)   F   
8    2012  Essential Hypertension and Renal Diseases (I10...   F   
9    2009     Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   
10   2011       Chronic Lower Respiratory Diseases (J40-J47)   M   
11   2008          Cerebrovascular Disease (Stroke: I60-I69)   F   
12   2013     Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
13   2010          Cerebrovascular Disease (Stroke: I60-I69)   F   
14   2012     Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
15   2012              Malignant Neoplasms (Cancer: C00-C97)   M   
16   2011          Cerebrovascular Disease (Stroke: I60-I69)   F   
17   2014  Congenital Malformations, Deformations, and Ch...   F   
18   2009                                   All Other Causes   F   
19   2014                        Diabetes Mellitus (E10-E14)   M   
20   2014  Accidents Except Drug Posioning (V01-X39, X43,...   M   
21   2012          Cerebrovascular Disease (Stroke: I60-I69)   F   
22   2009     Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
23   2014  Essential Hypertension and Renal Diseases (I10...   F   
24   2008  Certain Conditions originating in the Perinata...   M   
25   2012  Mental and Behavioral Disorders due to Acciden...   M   
26   2014              Malignant Neoplasms (Cancer: C00-C97)   F   
27   2011  Mental and Behavioral Disorders due to Acciden...   F   
28   2014  Accidents Except Drug Posioning (V01-X39, X43,...   F   
29   2008          Cerebrovascular Disease (Stroke: I60-I69)   M   
..    ...                                                ...  ..   
263  2012                               Septicemia (A40-A41)   F   
264  2014                                   All Other Causes   M   
265  2008                        Diabetes Mellitus (E10-E14)   M   
266  2008  Essential Hypertension and Renal Diseases (I10...   F   
267  2011  Human Immunodeficiency Virus Disease (HIV: B20...   F   
268  2012                        Diabetes Mellitus (E10-E14)   F   
269  2007    Intentional Self-Harm (Suicide: X60-X84, Y87.0)   M   
270  2009     Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
271  2014              Malignant Neoplasms (Cancer: C00-C97)   M   
272  2007                                   All Other Causes   M   
273  2014  Essential Hypertension and Renal Diseases (I10...   M   
274  2010     Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   
275  2008  Human Immunodeficiency Virus Disease (HIV: B20...   M   
276  2010            Influenza (Flu) and Pneumonia (J09-J18)   F   
277  2014     Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   
278  2014              Malignant Neoplasms (Cancer: C00-C97)   M   
279  2009                                   All Other Causes   M   
280  2012            Influenza (Flu) and Pneumonia (J09-J18)   F   
281  2008     Chronic Liver Disease and Cirrhosis (K70, K73)   F   
282  2013  Accidents Except Drug Posioning (V01-X39, X43,...   F   
283  2011       Chronic Lower Respiratory Diseases (J40-J47)   F   
284  2007                        Diabetes Mellitus (E10-E14)   F   
285  2012  Accidents Except Drug Posioning (V01-X39, X43,...   F   
286  2009                                   All Other Causes   M   
287  2013              Malignant Neoplasms (Cancer: C00-C97)   F   
288  2007                        Diabetes Mellitus (E10-E14)   M   
289  2010  Essential Hypertension and Renal Diseases (I10... 

<b><span style="color: #FF0000;"> Q1. Which year had the most deaths ? </span></b>

In [4]:
deaths.groupby('Year').count()

Leading Cause  Sex  Race Ethnicity  Deaths  Death Rate  \
Year                                                           
2007             37   37              37      37          37   
2008             31   31              31      31          31   
2009             35   35              35      35          35   
2010             38   38              38      38          38   
2011             43   43              43      43          43   
2012             42   42              42      42          42   
2013             28   28              28      28          28   
2014             39   39              39      39          39   

      Age Adjusted Death Rate  
Year                           
2007                       37  
2008                       31  
2009                       35  
2010                       38  
2011                       43  
2012                       42  
2013                       28  
2014                       39

<span style="color: #006600;"><b>Answer Q1. We can see 2011 had the most deaths with a total of 43.</b></span>

<span style="color: #FF0000;"><b> Q2.  Which sex had the highest death amount ?</b></span>

In [5]:
deaths.groupby('Race Ethnicity')['Sex'].count()

Race Ethnicity
Asian and Pacific Islander    46
Black Non-Hispanic            50
Hispanic                      50
Not Stated/Unknown            53
Other Race/ Ethnicity         49
White Non-Hispanic            45
Name: Sex, dtype: int64

<span style="color: #006600;"><b> Answer Q2. We can see the "Not Stated/Unknown" Race has the highest death toll with a total of 53, 
followed by "Black Non-Hispanic & Hipanics" at a total of 50.</b></span>

<span style="color: #FF0000;"><b> Q3. What were the top 5 leading causes of deaths  in 2007 ? </b></span>

In [36]:
deaths[deaths['Year']==2007].sort_values('Deaths', ascending = False).head(5)

Year                                      Leading Cause Sex  \
235  2007                                   All Other Causes   M   
182  2007                               Septicemia (A40-A41)   F   
98   2007          Cerebrovascular Disease (Stroke: I60-I69)   F   
163  2007          Cerebrovascular Disease (Stroke: I60-I69)   M   
178  2007  Accidents Except Drug Posioning (V01-X39, X43,...   M   

                 Race Ethnicity Deaths Death Rate Age Adjusted Death Rate  
235                    Hispanic    988       90.6                   142.6  
182          Black Non-Hispanic     89        8.5                     8.1  
98   Asian and Pacific Islander     70       13.6                    17.6  
163       Other Race/ Ethnicity      7          .                       .  
178  Asian and Pacific Islander     53         11                      13

<span style="color: #006600;"> <b>Answer Q3. We can see that the top (5) leading causes of death were Septicemia (89), Cerebrovascular Disease (77), Accidents Except Drug Posioning (53)and some other causes (988). </b></span>

<div align="center"><span style="color: #006600;"><h1> THANK YOU! </h1></span></div>